wagtail-rfcs/000-append-comments-for-moderation.md

```diff
@@ -1,53 +1,92 @@
+4. An update to the Admin UI's JS is required:
+
+   - The JS must watch for submission of a Page for moderation and prevent the HTTP POST that would normally send the form's payload to the server
+   - It will then trigger a basic modal popup to capture the required info from the Editor
+   - The modal will show the label, as described above
+   - The modal will have a button to complete the submission, as described above. It will take the `disabled` state while the `moderation_comment` `<textarea>`'s content has no length or is greater than 500 characters. The page's JS will update the `disabled` state of the submission button accordingly, so that only valid comment content can be submitted.
+   - Once the submission-completion button has been clicked, the full page PLUS the new moderation-request comment will be sent to the server. (How this is done is definitely open: it could be that the payload state is copied/held when the original POST is blocked, and then that payload is updated before finally being POSTed; alternatively, the whole page could be re-POSTed now the hidden `moderation_comment` field has been populated)
+
+5. The server receives the form payload and, upon seeing a `moderation_comment` field:
+
+   - validates that it is between 0 and 500 characters.
+   - ensures that data is saved with the `PageRevision` it creates.
+     - Note: If validation fails, Wagtail does not save the new revision and instead returns an error message using the same workflow as per any required field missing data.
+       - It might be useful to outline the Publish combo-button in red, to show where the user needs to go to amend the problem.
+
+6. When any Editor is informed via email that a Page has been submitted for moderation, the `moderation_comment` shall also be included (but must be marked safe/sanitised)
+7. When any Editor views the details of a Page pending Moderation (ie at `admin/pages/<pageid>/revisions/compare/live...<newrev>/`), if the relevant revision has a `moderation_comment`, it should be shown above the list of field diffs.
+
+Notes:
+
+- The same behaviour will need to be present on both page creation and page editing forms
+
+## Drawbacks
+
+- A code change will be needed to enable/disable the functionality as described above.
+- In some situations the functionality will not be needed, so this change adds complexity to provide non-core functionality.
+- This functionality will not guarantee an improved workflow: Editors can still circumvent it by typing nonsense into the comment field.
 
-### Drawbacks
-* A developer will be needed to deactivate the functionality as described above.
-* In some situations the functionality will not be needed.
-* When this functionality is implemented, it will be easy for editors to circumvent by typing nonsense into the comment field. 
+## Alternatives
 
-### Alternatives
-* More detailed linting functionality to enable administrators to ensure that comment submissions are not spurious. 
```

Comment:
@valgrimm I dropped this line because I noticed that Wagtail does have a useful way to see what has changed in a page, even if it it doesn't cover the _why_ of the change.

I wondered whether this would be useful (or even - at a long shot - enough)? 

![Uploading Screenshot 2019-12-30 at 13.27.10.png…]()
![Screenshot 2019-12-30 at 13 27 14](https://user-images.githubusercontent.com/101457/71587512-da009400-2b15-11ea-9fcb-18ac5ae250be.png)